diff --git a/NUnitLite-1.0.0/src/framework/Assert.cs b/NUnitLite-1.0.0/src/framework/Assert.cs
index 9f35e72..3c4f9a4 100755
--- a/NUnitLite-1.0.0/src/framework/Assert.cs
+++ b/NUnitLite-1.0.0/src/framework/Assert.cs
@@ -1157,6 +1157,41 @@ namespace NUnit.Framework
 
         #region AreEqual
 
+        #region IsInstanceOf
+
+        /// <summary>
+        /// Asserts that an object is an instance of a given type.
+        /// </summary>
+        /// <param name="expected">The expected Type</param>
+        /// <param name="actual">The object being examined</param>
+        /// <param name="message">The message to display in case of failure</param>
+        /// <param name="args">Array of objects to be used in formatting the message</param>
+        public static void IsInstanceOf(Type expected, object actual, string message, params object[] args)
+        {
+            Assert.That(actual, Is.InstanceOf(expected), message, args);
+        }
+        /// <summary>
+        /// Asserts that an object is an instance of a given type.
+        /// </summary>
+        /// <param name="expected">The expected Type</param>
+        /// <param name="actual">The object being examined</param>
+        /// <param name="message">The message to display in case of failure</param>
+        public static void IsInstanceOf(Type expected, object actual, string message)
+        {
+            Assert.That(actual, Is.InstanceOf(expected), message, null);
+        }
+        /// <summary>
+        /// Asserts that an object is an instance of a given type.
+        /// </summary>
+        /// <param name="expected">The expected Type</param>
+        /// <param name="actual">The object being examined</param>
+        public static void IsInstanceOf(Type expected, object actual)
+        {
+            Assert.That(actual, Is.InstanceOf(expected), null, null);
+        }
+
+        #endregion
+
         #region Ints
 
         /// <summary>
diff --git a/NUnitLite-1.0.0/src/framework/Internal/Commands/OneTimeSetUpCommand.cs b/NUnitLite-1.0.0/src/framework/Internal/Commands/OneTimeSetUpCommand.cs
index 5f0112b..5609ccf 100644
--- a/NUnitLite-1.0.0/src/framework/Internal/Commands/OneTimeSetUpCommand.cs
+++ b/NUnitLite-1.0.0/src/framework/Internal/Commands/OneTimeSetUpCommand.cs
@@ -56,7 +56,7 @@ namespace NUnit.Framework.Internal.Commands
         {
             if (fixtureType != null)
             {
-                if (context.TestObject == null && !IsStaticClass(fixtureType))
+                if ((context.TestObject == null || context.GetType() != fixtureType) && !IsStaticClass(fixtureType))
                     context.TestObject = Reflect.Construct(fixtureType, arguments);
 
                 foreach (MethodInfo method in  Reflect.GetMethodsWithAttribute(fixtureType, typeof(TestFixtureSetUpAttribute), true))
