diff -rupN -x .DS_Store -x .classpath -x .metadata -x .project main/java/com/couchbase/cblite/CBLManager.java sharpen/java/com/couchbase/cblite/CBLManager.java
--- main/java/com/couchbase/cblite/CBLManager.java	2013-10-14 17:37:49.000000000 -0700
+++ sharpen/java/com/couchbase/cblite/CBLManager.java	2013-10-14 17:37:21.000000000 -0700
@@ -14,9 +14,9 @@ import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
-public enum CBLManager {
+public class CBLManager {
 
-    INSTANCE;
+    //INSTANCE;
 
     private CBLServer server;
 
diff -rupN -x .DS_Store -x .classpath -x .metadata -x .project main/java/com/couchbase/cblite/CBLServer.java sharpen/java/com/couchbase/cblite/CBLServer.java
--- main/java/com/couchbase/cblite/CBLServer.java	2013-10-14 17:37:49.000000000 -0700
+++ sharpen/java/com/couchbase/cblite/CBLServer.java	2013-10-14 17:37:21.000000000 -0700
@@ -61,7 +61,7 @@ public class CBLServer {
     }
 
     public CBLServer(String directoryName) throws IOException {
-        this(directoryName, CBLManager.INSTANCE);
+        this(directoryName, new CBLManager());
     }
 
     public CBLServer(String directoryName, CBLManager manager) throws IOException {
diff -rupN -x .DS_Store -x .classpath -x .metadata -x .project main/java/com/couchbase/cblite/replicator/CBLReplicator.java sharpen/java/com/couchbase/cblite/replicator/CBLReplicator.java
--- main/java/com/couchbase/cblite/replicator/CBLReplicator.java	2013-10-14 17:37:49.000000000 -0700
+++ sharpen/java/com/couchbase/cblite/replicator/CBLReplicator.java	2013-10-14 17:37:21.000000000 -0700
@@ -120,7 +120,7 @@ public abstract class CBLReplicator exte
             }
         });
 
-        this.clientFacotry = clientFacotry != null ? clientFacotry : CBLHttpClientFactory.INSTANCE;
+        this.clientFacotry = clientFacotry != null ? clientFacotry : new CBLHttpClientFactory();
 
 
     }
diff -rupN -x .DS_Store -x .classpath -x .metadata -x .project main/java/com/couchbase/cblite/support/CBLHttpClientFactory.java sharpen/java/com/couchbase/cblite/support/CBLHttpClientFactory.java
--- main/java/com/couchbase/cblite/support/CBLHttpClientFactory.java	2013-10-14 17:37:49.000000000 -0700
+++ sharpen/java/com/couchbase/cblite/support/CBLHttpClientFactory.java	2013-10-14 17:37:21.000000000 -0700
@@ -15,9 +15,9 @@ import org.apache.http.params.BasicHttpP
 
 import java.util.List;
 
-public enum CBLHttpClientFactory implements HttpClientFactory {
+public class CBLHttpClientFactory implements HttpClientFactory {
 
-    INSTANCE;
+//    INSTANCE;
 
     private CookieStore cookieStore;
 
diff -rupN -x .DS_Store -x .classpath -x .metadata -x .project main/java/com/couchbase/cblite/support/CBLRemoteMultipartDownloaderRequest.java sharpen/java/com/couchbase/cblite/support/CBLRemoteMultipartDownloaderRequest.java
--- main/java/com/couchbase/cblite/support/CBLRemoteMultipartDownloaderRequest.java	2013-10-14 17:37:49.000000000 -0700
+++ sharpen/java/com/couchbase/cblite/support/CBLRemoteMultipartDownloaderRequest.java	2013-10-14 17:37:21.000000000 -0700
@@ -57,7 +57,7 @@ public class CBLRemoteMultipartDownloade
 
             // add in cookies to global store
             DefaultHttpClient defaultHttpClient = (DefaultHttpClient)httpClient;
-            CBLHttpClientFactory.INSTANCE.addCookies(defaultHttpClient.getCookieStore().getCookies());
+            new CBLHttpClientFactory().addCookies(defaultHttpClient.getCookieStore().getCookies());
 
 
             StatusLine status = response.getStatusLine();
diff -rupN -x .DS_Store -x .classpath -x .metadata -x .project main/java/com/couchbase/cblite/support/CBLRemoteRequest.java sharpen/java/com/couchbase/cblite/support/CBLRemoteRequest.java
--- main/java/com/couchbase/cblite/support/CBLRemoteRequest.java	2013-10-14 17:37:49.000000000 -0700
+++ sharpen/java/com/couchbase/cblite/support/CBLRemoteRequest.java	2013-10-14 17:37:21.000000000 -0700
@@ -114,7 +114,7 @@ public class CBLRemoteRequest implements
 
             // add in cookies to global store
             DefaultHttpClient defaultHttpClient = (DefaultHttpClient)httpClient;
-            CBLHttpClientFactory.INSTANCE.addCookies(defaultHttpClient.getCookieStore().getCookies());
+            new CBLHttpClientFactory().addCookies(defaultHttpClient.getCookieStore().getCookies());
 
             StatusLine status = response.getStatusLine();
             if (status.getStatusCode() >= 300) {
